### PR TITLE
16. T14 - Reduced string

### PR DESCRIPTION
**Pull request by sathya-git-tech**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,20 @@
-﻿namespace Training {
+﻿using System.Text;
+namespace Training {
    internal class Program {
       static void Main (string[] args) {
-         Console.WriteLine ("Hello World!");
+         Console.WriteLine ("Enter a string:");
+         string input = Console.ReadLine ();
+         char[] character = input.ToArray ();
```
Comment (priyarathna) [inline Program.cs]: This is not used anywhere in the program.

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,20 @@
-﻿namespace Training {
+﻿using System.Text;
+namespace Training {
    internal class Program {
       static void Main (string[] args) {
-         Console.WriteLine ("Hello World!");
+         Console.WriteLine ("Enter a string:");
+         string input = Console.ReadLine ();
+         char[] character = input.ToArray ();
+         input = input.ToLower ();
```
Comment (priyarathna) [inline Program.cs]: Can be included like string input = = Console.ReadLine ().ToLower ();

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
+﻿using System.Text;
+namespace Training {
    internal class Program {
       static void Main (string[] args) {
-         Console.WriteLine ("Hello World!");
+         Console.WriteLine ("Enter a string:");
+         string input = Console.ReadLine ();
+         char[] character = input.ToArray ();
+         input = input.ToLower ();
+         StringBuilder result = new StringBuilder ();
```
Comment (priyarathna) [inline Program.cs]: Include like this: StringBuilder result = new ();

File: Program.cs [hunk 4]
```diff
@@ -1,7 +1,20 @@
...
+
```
Comment (priyarathna) [inline Program.cs]: No need of empty line here.

File: Program.cs [hunk 5]
```diff
@@ -1,7 +1,20 @@
...
+         for (int i = 0; i < input.Length; i++) {
+            if (i == input.Length - 1 || input[i] != input[i + 1]) {
```
Comment (priyarathna) [inline Program.cs]: Why the check "i == input.Length - 1" is needed? And for single if statements, don't include flower brackets ('{').

File: Program.cs [hunk 6]
```diff
@@ -1,7 +1,20 @@
...
+               result.Append (input[i]);
+            } 
+            else i++;
+         }
+         Console.WriteLine (result.Length==0? "Reduced string is 'Empty'" : $"Reduced string is '{result}'");
```
Comment (priyarathna) [inline Program.cs]: This line is not aligned. Use "Ctrl+K & Ctrl+F" to see the difference.

File: Program.cs [hunk 7]
```diff
@@ -1,7 +1,20 @@
...
    internal class Program {
       static void Main (string[] args) {
-         Console.WriteLine ("Hello World!");
+         Console.WriteLine ("Enter a string:");
+         string input = Console.ReadLine ();
+         char[] character = input.ToArray ();
+         input = input.ToLower ();
+         StringBuilder result = new StringBuilder ();
+
+         for (int i = 0; i < input.Length; i++) {
```
Comment (priyarathna) [inline Program.cs]: Store input.Length in local variable and use it.

File: Program.cs (hunk 5 above)
Comment (sathya-git-tech) [reply Program.cs]: For this code input[i] != input[i + 1], the program will crash for the last character as there is no character after that. So to include the last character in the loop, i == input.Length - 1 is used.

**Review by priyarathna — APPROVED** [review]

**Pull request closed without merging**